src/XMonad/ManageHook.hs

```diff
@@ -106,6 +106,14 @@ getStringProperty d w p = do
   md <- io $ getWindowProperty8 d a w
   return $ fmap (map (toEnum . fromIntegral)) md
 
+-- | Return whether the window will be a floating window or not
+willFloat :: Query Bool
+willFloat = ask >>= \w -> liftX $ withDisplay $ \d -> do
```

Comment:
Would it make more sense to refactor the code in XMonad.Operations to separate out its check, and then use that here?

Reply:
It totally does, I did just that in my last commit. Thanks for pointing it out, I completely missed it the first time around :slightly_smiling_face: